lib/readline.js

```diff
@@ -474,8 +491,60 @@ Interface.prototype._insertString = function(c) {
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+
+  searchHistory.call(this);
 };
 
+function appendSearchResult(result) {
+  // this.previewResult = result;
+
+  // Cursor to left edge.
+  cursorTo(this.output, 0);
+  clearScreenDown(this.output);
+
+  // Based on the line and match result
+  // write the data
+  if (result !== '') {
+    this.output.write(buildReverseSearchPrompt(this.line, result));
+    cursorTo(this.output, this.cursor + kReverseSearchPrompt.length - 2);
+  } else if (this.line === '') {
+    this.output.write(buildReverseSearchPrompt());
+    cursorTo(this.output, this.cursor + kReverseSearchPrompt.length - 2);
```

Comment:
We could move the cursor to the position where we matched the entry. That way we also use the same behavior as the bash and it's helpful for the user.

Reply:
+1. I guess this is very good idea!